httpe/err.go

```diff
@@ -0,0 +1,85 @@
+	ErrNotAcceptable                = StatusError(http.StatusNotAcceptable)
+	ErrProxyAuthRequired            = StatusError(http.StatusProxyAuthRequired)
+	ErrRequestTimeout               = StatusError(http.StatusRequestTimeout)
+	ErrConflict                     = StatusError(http.StatusConflict)
+	ErrGone                         = StatusError(http.StatusGone)
+	ErrLengthRequired               = StatusError(http.StatusLengthRequired)
+	ErrPreconditionFailed           = StatusError(http.StatusPreconditionFailed)
+	ErrRequestEntityTooLarge        = StatusError(http.StatusRequestEntityTooLarge)
+	ErrRequestURITooLong            = StatusError(http.StatusRequestURITooLong)
+	ErrUnsupportedMediaType         = StatusError(http.StatusUnsupportedMediaType)
+	ErrRequestedRangeNotSatisfiable = StatusError(http.StatusRequestedRangeNotSatisfiable)
+	ErrExpectationFailed            = StatusError(http.StatusExpectationFailed)
+	ErrTeapot                       = StatusError(http.StatusTeapot)
+	ErrMisdirectedRequest           = StatusError(http.StatusMisdirectedRequest)
+	ErrUnprocessableEntity          = StatusError(http.StatusUnprocessableEntity)
+	ErrLocked                       = StatusError(http.StatusLocked)
+	ErrFailedDependency             = StatusError(http.StatusFailedDependency)
+	ErrTooEarly                     = StatusError(http.StatusTooEarly)
+	ErrUpgradeRequired              = StatusError(http.StatusUpgradeRequired)
+	ErrPreconditionRequired         = StatusError(http.StatusPreconditionRequired)
+	ErrTooManyRequests              = StatusError(http.StatusTooManyRequests)
+	ErrRequestHeaderFieldsTooLarge  = StatusError(http.StatusRequestHeaderFieldsTooLarge)
+	ErrUnavailableForLegalReasons   = StatusError(http.StatusUnavailableForLegalReasons)
+
+	ErrInternalServerError           = StatusError(http.StatusInternalServerError)
+	ErrNotImplemented                = StatusError(http.StatusNotImplemented)
+	ErrBadGateway                    = StatusError(http.StatusBadGateway)
+	ErrServiceUnavailable            = StatusError(http.StatusServiceUnavailable)
+	ErrGatewayTimeout                = StatusError(http.StatusGatewayTimeout)
+	ErrHTTPVersionNotSupported       = StatusError(http.StatusHTTPVersionNotSupported)
+	ErrVariantAlsoNegotiates         = StatusError(http.StatusVariantAlsoNegotiates)
+	ErrInsufficientStorage           = StatusError(http.StatusInsufficientStorage)
+	ErrLoopDetected                  = StatusError(http.StatusLoopDetected)
+	ErrNotExtended                   = StatusError(http.StatusNotExtended)
+	ErrNetworkAuthenticationRequired = StatusError(http.StatusNetworkAuthenticationRequired)
+)
+
+// WriteHTTPeErr writes the http status represented by err to the ResponseWriter and
+// prints the error message for 4xx errors, but keeps details for 5xx errors.
+func WriteHTTPeErr(w http.ResponseWriter, err error) {
+	if err == nil {
+		return
+	}
+	sErr := StatusError(0)
```

Comment:
suggestion for simpler alternate implementation:
```
        if err == nil {
                return
        }
        sErr := ErrInternalServerError
        if ok := errors.As(err, &sErr); !ok || sErr.IsInternal() {
                // Hide the actual error to prevent information leakage
                err = sErr
        }
        http.Error(w, err.Error(), sErr.Code())
```

Reply:
ok